### PR TITLE
Prevent too long identifier names

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref #10213

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Fails for the calendar:

```
Index name "oc_calendar_resources_cache"."calendar_resources_cache_id_idx" is too long.
```

**Comment by nickvergessen** [comment]
Yeah wanted to fail it and now rebasing on https://github.com/nextcloud/server/pull/10213

**Review by MorrisJobke — APPROVED** [review]
Tested and works now 👍 Also code makes sense

**Comment by rullzer** [comment]
Can't this still do :boom: is people chose a to long prefix?

**Comment by georgehrke** [comment]
> Can't this still do :boom: is people chose a to long prefix?

Well, that's the case for any table, not only this resource booking ones. If someone selects a 30-char long prefix it will just go :boom:
We should properly document this though cc @MariusBluem

**Review by georgehrke — APPROVED** [review]

File: lib/private/DB/MigrationService.php [hunk 1]
```diff
@@ -457,6 +460,43 @@ public function executeStep($version) {
 		$this->markAsExecuted($version);
 	}
 
+	public function ensureOracleIdentifierLengthLimit(Schema $schema) {
+		foreach ($schema->getTables() as $table) {
```
Comment (MorrisJobke) [inline lib/private/DB/MigrationService.php]:
```
1) Test\DB\MigrationsTest::testExecuteStepWithSchemaChange
Invalid argument supplied for foreach()

/drone/src/github.com/nextcloud/server/lib/private/DB/MigrationService.php:464
/drone/src/github.com/nextcloud/server/lib/private/DB/MigrationService.php:451
/drone/src/github.com/nextcloud/server/tests/lib/DB/MigrationsTest.php:120
```

**Comment by nickvergessen** [comment]
> Can't this still do boom is people chose a to long prefix?

Well if you use oracle you usually dont change such stuff anyway, because you know the limitations. The current code makes sure it works with a default prefix (3 chars). Anything else is not our business

File: lib/private/DB/MigrationService.php (hunk 1 above)
Comment (nickvergessen) [reply lib/private/DB/MigrationService.php]: Fixed and also added unit tests for all the checks

**Comment by MorrisJobke** [comment]
```
Sequence name "oc_twofactor_backupcodes_id_seq" is too long.
```

**Comment by nickvergessen** [comment]
Right, default name on postgres is also not `PRIMARY`.
Let's see

**Comment by MorrisJobke** [comment]
`Sequence name "oc_calendarsubscriptions_id_seq" is too long.`

**Comment by nickvergessen** [comment]
Postgres is so annoying, anyway pushed a fix

**Comment by MorrisJobke** [comment]
```
Sequence name "oc_twofactor_backupcodes_id_seq" is too long.  
```

**Comment by nickvergessen** [comment]
I shall set it up locally for faster debugging this week.

**Comment by nickvergessen** [comment]
Fixed and tested locally with postgres, mysql and sqlite

**Review by rullzer — APPROVED** [review]
:elephant:

**Pull request merged**